### PR TITLE
Rhat using split chains

### PR DESCRIPTION
**Pull request by ben18785**
As per #1061. Have compared with Stan too and is the same.

### CODE REVIEW

**Comment by ben18785** [comment]
Also converting ess estimates to be the same as how Stan calculates them.

**Comment by ben18785** [comment]
@martinjrobins I've decided to split out the ess calculations bit into another PR as it's quite a bit.

File: pints/_diagnostics.py [hunk 1]
```diff
@@ -96,9 +101,12 @@ def reorder_all_params(chains):
...
 def rhat(samples):
-    """
-    Calculates r-hat = sqrt(((n - 1)/n * W + (1/n) * B)/W) as per "Bayesian
-    data analysis", 3rd edition, Gelman et al., 2014.
+    r"""
+    Calculates :math:`\hat{R} = sqrt(((n - 1)/n * W + (1/n) * B)/W)`` as per
+    [1] for a single parameter. It does this after splitting each chain into
+    two.
+
+    [1] "Bayesian data analysis", 3rd edition, Gelman et al., 2014.
```
Comment (MichaelClerx) [inline pints/_diagnostics.py]: Update to
```
    References
    ----------
    .. [1] "Bayesian...
```

File: pints/_diagnostics.py [hunk 2]
```diff
@@ -107,11 +115,12 @@ def rhat(samples):
...
 
 def rhat_all_params(chains):
-    """
-    Calculates r-hat for all parameters in chains as per "Bayesian data
-    analysis", 3rd edition, Gelman et al., 2014.
+    r"""
+    Calculates :math:`\hat{R} = sqrt(((n - 1)/n * W + (1/n) * B)/W)`` as per
+    [1] for all parameters. It does this after splitting each chain into two.
+
+    [1] "Bayesian data analysis", 3rd edition, Gelman et al., 2014.
```
Comment (MichaelClerx) [inline pints/_diagnostics.py]: See above

File: pints/_diagnostics.py (hunk 1 above)
Comment (ben18785) [reply pints/_diagnostics.py]: Done

File: pints/_diagnostics.py (hunk 2 above)
Comment (ben18785) [reply pints/_diagnostics.py]: Done!

**Comment by MichaelClerx** [comment]
> Also converting ess estimates to be the same as how Stan calculates them.

Are you... _stanning_ Stan?

**Pull request merged**